src/flatland/ordered/set.clj

```diff
@@ -9,13 +9,6 @@
 
 (declare transient-ordered-set)
 
-;; We could use compile-if technique here, but hoping to avoid
-;; an AOT issue using this way instead.
-(def hasheq-ordered-set
-  (or (resolve 'clojure.core/hash-unordered-coll)
```

Comment:
This is actually OK on the top level

Reply:
I thought there was a similar case as with the ordered map, a runtime require, but that's not true it seems

Reply:
So there is no need for this PR and the bb PR seems good then

Reply:
Cool, glad to hear the bb PR is unblocked. I think this is a worthwhile change anyway, given that we no longer support the versions the workaround exists for, but I'll wait for some others to take a peek before merging.